appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/receiver/service/ReceiveKinServiceBase.java

```diff
@@ -0,0 +1,108 @@
+    private static final String SERVICE_ARG_TO_ADDRESS = "toAddress";
+    private static final String SERVICE_ARG_AMOUNT = "amount";
+    private static final String SERVICE_ARG_TRANSACTION_ID = "transactionId";
+    private static final String SERVICE_ARG_MEMO = "memo";
+    private static final String SERVICE_ARG_ERROR = "error";
+    private static final int NO_AMOUNT = 0;
+
+
+    public static void notifyTransactionCompleted(@NonNull Context context, @NonNull String fromAddress, @NonNull String toAddress, int amount, @NonNull String transactionId, @NonNull String memo) {
+        Intent intent = getTransactionResultIntent(context, true);
+        if (intent != null) {
+            intent.putExtra(SERVICE_ARG_FROM_ADDRESS, fromAddress);
+            intent.putExtra(SERVICE_ARG_TO_ADDRESS, toAddress);
+            intent.putExtra(SERVICE_ARG_AMOUNT, amount);
+            intent.putExtra(SERVICE_ARG_TRANSACTION_ID, transactionId);
+            intent.putExtra(SERVICE_ARG_MEMO, memo);
+            context.startService(intent);
+        }
+    }
+
+    public static void notifyTransactionFailed(@NonNull Context context, @NonNull String error, @NonNull String fromAddress, @NonNull String toAddress, int amount, @NonNull String memo) {
+        Intent intent = getTransactionResultIntent(context, false);
+        if (intent != null) {
+            intent.putExtra(SERVICE_ARG_ERROR, error);
+            intent.putExtra(SERVICE_ARG_FROM_ADDRESS, fromAddress);
+            intent.putExtra(SERVICE_ARG_TO_ADDRESS, toAddress);
+            intent.putExtra(SERVICE_ARG_AMOUNT, amount);
+            intent.putExtra(SERVICE_ARG_MEMO, memo);
+            context.startService(intent);
+        }
+    }
+
+    private static Intent getTransactionResultIntent(Context context, Boolean isCompleted) {
+        String action = isCompleted ? ACTION_TRANSACTION_COMPLETED : ACTION_TRANSACTION_FAILED;
+        Intent intent = new Intent();
+        intent.setAction(action);
+
+        //TODO change to receiver pkg
+        final String receiverPackageName = context.getApplicationContext().getPackageName();
+
+        intent.setComponent(new ComponentName(receiverPackageName, receiverPackageName + ".ReceiveKinService"));
+        intent.setPackage(receiverPackageName);
+        final List<ResolveInfo> resolveInfos = context.getPackageManager().queryIntentServices(intent, 0);
+        if (resolveInfos.isEmpty()) {
+            //TODO notify error service not found - hence the receiver will not be inform of the transaction
+            Log.d("####", "#### cant find the service " + receiverPackageName + ".ReceiveKinService");
+        } else {
+            for (ResolveInfo info : resolveInfos) {
+                if (info.serviceInfo.exported) {
+                    //TODO throw exception service is exported
+                    Log.d("####", "####  service " + receiverPackageName + ".ReceiveKinService export must be declared on manifest false");
```

Comment:
isn't `exported` is what we **do** looking for?

Reply:
no it should not be exported so other apps can use it